kratos/processes/integration_values_extrapolation_to_nodes_process.cpp

```diff
@@ -43,26 +43,26 @@ IntegrationValuesExtrapolationToNodesProcess::IntegrationValuesExtrapolationToNo
     mAreaAverage = ThisParameters["area_average"].GetBool();
 
     // The average variable
-    mAverageVariable = KratosComponents< Variable<double> >::Get(ThisParameters["average_variable"].GetString());
+    mpAverageVariable = &(const_cast<Variable<double>&>(KratosComponents<Variable<double>>::Get(ThisParameters["average_variable"].GetString())));
 
     // We get the list of variables
     const SizeType n_variables = ThisParameters["list_of_variables"].size();
 
-    for (IndexType i_var = 0; i_var < n_variables; ++i_var){
-        const std::string& variable_name = ThisParameters["list_of_variables"].GetArrayItem(i_var).GetString();
-
-        if(KratosComponents<Variable<double>>::Has(variable_name)){
-            Variable<double> variable = KratosComponents< Variable<double> >::Get(variable_name);
-            mDoubleVariable.push_back(variable);
-        } else if (KratosComponents< Variable< array_1d< double, 3> > >::Has(variable_name)) {
-            Variable<array_1d< double, 3>> variable = KratosComponents< Variable<array_1d< double, 3>> >::Get(variable_name);
-            mArrayVariable.push_back(variable);
-        } else if (KratosComponents< Variable< Vector > >::Has(variable_name)) {
-            Variable<Vector> variable = KratosComponents< Variable<Vector> >::Get(variable_name);
-            mVectorVariable.push_back(variable);
-        } else if (KratosComponents< Variable< Matrix > >::Has(variable_name)) {
-            Variable<Matrix> variable = KratosComponents< Variable<Matrix> >::Get(variable_name);
-            mMatrixVariable.push_back(variable);
+    for (IndexType p_var = 0; p_var < n_variables; ++p_var){
+        const std::string& r_variable_name = ThisParameters["list_of_variables"].GetArrayItem(p_var).GetString();
+
+        if (KratosComponents<Variable<double>>::Has(r_variable_name)){
+            const Variable<double>& r_variable = KratosComponents< Variable<double>>::Get(r_variable_name);
+            mDoubleVariable.push_back(&(const_cast<Variable<double>&>(r_variable)));
```

Comment:
why extracting as const and then const_cast?

cannot you do `mDoubleVariable.push_back(KratosComponents< Variable<double>>::Get(r_variable_name));`?

Reply:
That means your vector is fill with constant references, you cannot declare a vector of constant references, you can do it with pointers

Reply:
cannot you do `mDoubleVariable.push_back(*(KratosComponents< Variable<double>>::Get(r_variable_name)));` then?

Reply:
what I don't get is why you get as const and then const_cast in the next line

Reply:
Does not compile (cannot convert const Variable<double>& to Variable<double>* without removing const). I think in this case is safe to drop the const qualifier, as long as we only use the variables, we don't change them

Reply:
> what I don't get is why you get as const and then const_cast in the next line

To convert it into a pointer, I can do it all at the same time, but the line gets quite long...

Reply:
how abt the following then?
~~~c++
auto& r_variable = KratosComponents< Variable<double>>::Get(r_variable_name);
mDoubleVariable.push_back(&r_variable);
~~~

Reply:
Does not work with dropping the const (I already tried)

Reply:
ok it really does not work :)